internal/command/machine/run.go

```diff
@@ -659,7 +667,7 @@ func determineMachineConfig(ctx context.Context, initialMachineConf api.MachineC
 		machineConf.Init.Entrypoint = splitted
 	}
 
-	if cmd := flag.Args(ctx)[1:]; len(cmd) > 0 {
+	if cmd := flag.Args(ctx)[1:]; len(cmd) > 0 && len(machineConf.Init.Cmd) == 0 {
```

Comment:
hmm, the check on `Init.Cmd` length is preventing the update if already set, is that the expected behavior?

Reply:
Yeah. If I'm reading the code correctly, `determineMachineConfig` is used from both `fly m run` and `fly m update`. `run` accepts its command as a positional argument while `update` accepts it via flag. I was trying to ensure that the command wouldn't be trashed, and since these methods of injecting a command are mutually exclusive, the cleanest way to not trash the configured command seemed to be checking for an existing configuration. Otherwise `fly m update /bin/ls` might work when it shouldn't.

Unless Cobra is smart enough to not pass the positional arg through based on the docstring? Haven't figured out if that is a thing yet.

Thanks again.

Reply:
I added a comment clarifying that check. Arguably it might need to be rearchitected to accept the command as a separate argument to `determineMachineConfig`--I'm perpetually faced with choosing between refactoring or preserving, so please do let me know if I land on the wrong side of that choice. :)

Reply:
Considering all the other flags in use in this function I think we "preserve" and let your comment live like a _HERE BE DRAGONS_ warning.